addons/mail/static/src/model/model_field.js

```diff
@@ -792,97 +656,81 @@ class ModelField {
-            // other record may be deleted due to causality, avoid useless
-            // processing
-            if (otherRecord) {
-                // unlink current record from other record
-                otherRecord.update({
-                    [this.inverse]: [['unlink', record]],
-                });
-            }
-        }
+        // register dependents
+        this.write(record, otherRecords);
     }
 
     /**
-     * Handling of a `set` 'unlink' of a many2one relational field.
+     * Handling of a `set` 'unlink' of a x2one relational field.
      *
      * @private
      * @param {mail.model} record
      */
-    _setRelationUnlinkMany2One(record) {
-        const otherLocalId = this.read(record);
-        const OtherModel = this.env.models[this.to];
-
+    _setRelationUnlinkX2One(record) {
+        const deleteOtherRecord = this.read(record);
         // other record already unlinked, avoid useless processing
-        if (!otherLocalId) {
+        if (!deleteOtherRecord) {
             return;
         }
-
         // unlink other record from current record
         this.write(record, undefined);
-
-        const otherRecord = OtherModel.get(otherLocalId);
-        // other record may be deleted due to causality, avoid useless
-        // processing
-        if (otherRecord) {
-            // unlink current record from other record
-            otherRecord.update({
-                [this.inverse]: [['unlink', record]],
-            });
+        // unlink current record from other record
+        deleteOtherRecord.update({
+            [this.inverse]: [['unlink', record]],
+        });
+        // apply causality
+        if (this.isCausal) {
+            deleteOtherRecord.delete();
         }
     }
 
     /**
-     * Handling of a `set` 'unlink' of an one2one relational field.
+     * Verifies the given relational value makes sense for the current field.
      *
      * @private
      * @param {mail.model} record
+     * @throws {Error} if record does not satisfy related model
      */
-    _setRelationUnlinkOne2One(record) {
-        const otherLocalId = this.read(record);
-        const OtherModel = this.env.models[this.to];
-
-        // other record already unlinked, avoid useless processing
-        if (!otherLocalId) {
+    _verifyRelationalValue(record) {
+        if (record === undefined) {
+            // ignored value, often result of compute methods
             return;
         }
-
-        // unlink other record from current record
-        this.write(record, undefined);
-
-        const otherRecord = OtherModel.get(otherLocalId);
-        // other record may be deleted due to causality, avoid useless
-        // processing
-        if (otherRecord) {
-            // unlink current record from other record
-            otherRecord.update({
-                [this.inverse]: [['unlink', record]],
-            });
+        if (record === null) {
+            // "unlink-all" value
+            return;
+        }
```

Comment:
Can't be removed? We don't want to support `null` as `unlink-all` this anymore.

Reply:
Probably yes, I didn't think this was the intent.